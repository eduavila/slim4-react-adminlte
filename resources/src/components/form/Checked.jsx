import React from 'react';
import { Checkbox } from 'react-icheck';
import { toCssClasse } from '../../helpers/utils';

const Checked = ({
  field, // { name, value, onChange, onBlur }
  form: { touched, errors,setFieldValue }, // also values, setXXXX, handleXXXX, dirty, isValid, status, etc.
  ...props
}) => {

    return (
        <div className={`form-group ${toCssClasse(props.cols || `12`)} has-feedback ${touched[field.name] && errors[field.name] ? "has-error" : ""}`}>
            <label className="control-label" style={{display:'block'}}>
                {props.description || ''}
                {props.isRequired &&
                <span style={{ color: 'red' }} title="Campo ObrigatÃ³rio"><b>*</b></span>}
                { props.helpDescription && 
                    <i className="fa fa-question-circle" title={ props.helpDescription }></i> 
                }
            </label>

            <Checkbox
                {...field}
                {...props}
                id={field.name}
                checkboxClass={`icheckbox_flat-blue`}
                disabled={props.readOnly}
                disabledClass='readonly'
                increaseArea="15%"
                value={field.checkedValue ? checkedValue : false}
                checked={field.value}
                onChange={e => setFieldValue(field.name,!field.value)}
                className={`form-control ${touched[field.name] && errors[field.name] ? "is-invalid" : ""}`}
            />

            { touched[field.name] &&
                errors[field.name] && <div className="help-block">{errors[field.name]}</div>}
        </div>
    )
};

export default Checked;