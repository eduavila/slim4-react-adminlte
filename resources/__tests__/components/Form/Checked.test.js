import React from 'react';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import Checked from '../../../src/components/form/Checked';

describe('<Checked />', () => {
    const mockSetFieldValue = jest.fn();
    const mockSetFieldTouched = jest.fn();

    const mockProps = {
        field:{
            name: 'check_mock',
            value: false
        },
        form: { 
            touched: {}, 
            errors: {},
            setFieldValue: mockSetFieldValue, 
            setFieldTouched: mockSetFieldTouched
        },
        props:{
            descrition: 'Campo de mock:'
        }
    }

    describe('render()', () => {
        test('renders the component', () => {
            const wrapper = shallow(<Checked { ...mockProps }/>);
        
            expect(toJson(wrapper)).toMatchSnapshot();
        });

        test('click para checkar Checked', () => {
            const wrapper = mount(<Checked { ...mockProps }/>);

            wrapper.find('#check_mock').simulate('change', { target: { value: true } });

            expect(mockSetFieldValue.mock.calls.length).toEqual(1);
            expect(mockSetFieldTouched.mock.calls.length).toEqual(1);

            expect(mockSetFieldValue.mock.calls[0][1]).toEqual(true);
        });

        test('Message de erro no Checked', () => {
            const mockForm = {
                ...mockProps.form,
                touched: {
                    check_mock: true
                }, 
                errors: {
                    check_mock:'Teste de erro.'
                }
            }
            const wrapper = mount(<Checked field={mockProps.field} {...mockProps.props} form={mockForm}/>);

            const text = wrapper.find('.help-block').text();
            expect(text).toEqual(mockForm.errors.check_mock);
        });
    });
});
