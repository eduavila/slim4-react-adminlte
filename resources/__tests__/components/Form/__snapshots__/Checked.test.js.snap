// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Checked /> render() renders the component 1`] = `
<div
  className="form-group  col-xs-12 has-feedback "
>
  <label
    className="control-label"
    style={
      Object {
        "display": "block",
      }
    }
  />
  <Checkbox
    checkboxClass="icheckbox_flat-blue"
    checked={false}
    className="form-control "
    disabledClass="readonly"
    id="checkbox1"
    increaseArea="15%"
    name="check_mock"
    onChange={[Function]}
    props={
      Object {
        "descrition": "Campo de mock:",
      }
    }
    value={false}
  />
</div>
`;
